report/src/main/java/com/hartwig/actin/report/pdf/tables/clinical/PatientClinicalHistoryGenerator.java

```diff
@@ -181,18 +168,31 @@ private static String secondPrimaryHistory(@NotNull ClinicalRecord record) {
         }
     }
 
-    @Nullable
-    private static String toDateString(@Nullable Integer year, @Nullable Integer month) {
+    @NotNull
+    private static Optional<String> toDateRangeString(@NotNull PriorTumorTreatment priorTumorTreatment) {
+        Optional<String> startOption = toDateString(priorTumorTreatment.startYear(), priorTumorTreatment.startMonth());
+        Optional<String> stopOption = toDateString(priorTumorTreatment.stopYear(), priorTumorTreatment.stopMonth());
+        return startOption.map(startString -> startString + stopOption.map(stopString -> "-" + stopString).orElse(""))
+                .or(() -> stopOption.map(stopString -> "end: " + stopString));
+    }
+
+    @NotNull
+    private static Optional<String> toDateString(@Nullable Integer year, @Nullable Integer month) {
         if (year != null) {
-            return month != null ? month + "/" + year : String.valueOf(year);
+            return month != null ? Optional.of(month + "/" + year) : Optional.of(String.valueOf(year));
         } else {
-            return null;
+            return Optional.empty();
         }
     }
 
-    @Nullable
-    private static String toStopReasonString(@Nullable String stopReason) {
-        return stopReason != null ? "stop reason: " + stopReason : null;
+    @NotNull
+    private static Optional<String> toNumberOfCyclesString(@Nullable Integer cycles) {
+        return cycles != null ? Optional.of(cycles + " cycles") : Optional.empty();
```

Comment:
You can use Optional.ofNullable().map() here instead.

Reply:
Yes, that's better.